### DISCUSSION TITLE
React Query keys no longer exported?

### DISCUSSION THREAD

**Discussion by tlqvst** [orval-labs/orval#1436] [General]
After updating from 6.24.0 to 6.29.1, all react query keys in the generated code are no longer exported. I couldn't find antything about this in the documentation, is there a way to enable this again? Or am I just misunderstanding how to handle query invalidations now? I often find myself needing the generated query keys.

**Comment by melloware** [comment]
I am using 6.29.1 and it looks like the keys are being exported?

Example:

```js
export const getGetEntityCarsQueryKey = (params?: GetEntityCarsParams) => {
	return [`/entity/cars`, ...(params ? [params] : [])] as const;
};
```

**Reply by tlqvst** [reply]
Seems like maybe it only happens with queries that are listed in the orval config with an overriden query type, e.g.

```js
output: {
    client: 'react-query',
    override: {
        operations: {
            MyOperation: {
                query: {
                    useQuery: true
                },
            },
        },
    },
}
```

**Reply by melloware** [reply]
If that is the case then can you open a bug with a reproducer?

**Reply by tlqvst** [reply]
Sure, just wanted to make sure it wasn't an intentional change that I missed somewhere. Thanks!

**Reply by melloware** [reply]
Not that I am aware of.

**Reply by tlqvst** [reply]
So I found this

https://github.com/anymaniax/orval/commit/3d965f4e1b1643815901cd0c3dba98d6cde50d14

There is an option called `shouldExportQueryKey`, but I can't figure out if it's intended behavior that omitting it should in fact exclude it..

**Reply by melloware** [reply]
I would ask the developer on https://github.com/anymaniax/orval/pull/1333

**Comment by tlqvst** [comment]
Made a codesandbox
https://codesandbox.io/p/devbox/xy6ffl

Check the `getListPetsQueryKey` function in `pets.ts`, then remove the `override` section in `orval.config.cjs` and run `pnpm generate-api`, notice that `getListPetsQueryKey` is now being exported.

**Reply by melloware** [reply]
@tlqvst can you try 6.30.2 and report back?

**Reply by tlqvst** [reply]
Yep, it's fixed! Thanks for helping out
